lib/ansible/cli/doc.py

```diff
@@ -595,7 +595,7 @@ def _list_plugins(self, plugin_type, loader):
         if len(context.CLIARGS['args']) == 1:
             coll_filter = context.CLIARGS['args'][0]
 
-        if coll_filter in ('', None):
+        if coll_filter in ('ansible.builtin', '', None):
```

Comment:
also missing ansible.legacy

also changelog and tests

Reply:
I pushed a new version. I failed to write a test because [_list_plugins()](https://github.com/ansible/ansible/blob/devel/lib/ansible/cli/doc.py#L591) needs a functional `ansible.plugins.loader`.

Reply:
You can import one from ansible.plugins.loader. For example
```python
from ansible.cli.doc import DocCLI
from ansible.plugins.loader import module_loader

def test_namespace_docs():
    args = ['ansible-doc', '-l', 'ansible.builtin', '-t', 'module']
    obj = DocCLI(args=args)
    super(DocCLI, obj).run()
    result = obj._list_plugins('module', module_loader)
    ...
```

Alternatively, there's also an integration test suite for `ansible-doc`. The entrypoint is the runme.sh file:  https://github.com/ansible/ansible/blob/devel/test/integration/targets/ansible-doc/runme.sh. You could just add a couple more `ansible-doc` commands to list builtin/legacy and test the output looks as expected. If you haven't run integration tests before: https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html#non-destructive-tests

Reply:
Thanks, that was the help I needed. I included the tests now.